packages/stream-metadata/src/environment.ts

```diff
@@ -27,10 +27,15 @@ function makeConfig() {
 	// eslint-disable-next-line no-process-env -- this is the only line where we're allowed to use process.env
 	const env = envSchema.parse(process.env)
 	const web3Config = getWeb3Deployment(env.RIVER_ENV)
-	const riverStreamMetadataBaseUrl = new URL(env.RIVER_STREAM_METADATA_BASE_URL)
+	const baseUrl = new URL(env.RIVER_STREAM_METADATA_BASE_URL)
 	if (env.RIVER_ENV.startsWith('local')) {
-		riverStreamMetadataBaseUrl.port = env.PORT.toString()
+		baseUrl.port = env.PORT.toString()
 	}
+	// Ensure no trailing slash when converting to string
+	const riverStreamMetadataBaseUrl = `${baseUrl.origin}${baseUrl.pathname.replace(
+		/\/$/,
+		'',
+	)}`.toLowerCase()
```

Comment:
@tak-hntlabs `baseUrl.origin` by itself should do the trick.

also ideally the source code doesn't deal with any of this, and it just takes the base url as an environment variable - localhost or not

Reply:
no it doesn't. The test caught the bug. `.origin` adds a trailing / - which is why I cut it out.

Reply:
@tak-hntlabs the issue you're seeing is not coming from `origin`, it's coming from `pathname`. `origin` does not even have a concept of a path to begin with, let alone a trailing slash. you can replace lines 35-38 with the following:

`const riverStreamMetadataBaseUrl` = baseUrl.origin;

For reference:
<img width="554" alt="image" src="https://github.com/user-attachments/assets/40ea287e-8e33-4642-8dab-0362e2a3a3be">

That will get the job done, but i also think the job you're trying to accomplish itself does not belong here. The environment sets the BASE_URL. if the BASE_URL should not have a trailing slash, we can just make it so by setting the correct environment variable. I surmise you're trying to cover development edge cases here but it's polluting the code we're shipping to prod. you can just set the env var in your package.json scripts and call it

Reply:
are you looking at the latest?